volcli/volcli.go

```diff
@@ -567,3 +572,73 @@ func useTheForce(ctx *cli.Context) (bool, error) {
 
 	return false, nil
 }
+
+// UseExec acquires a lock (waiting if necessary) and executes a command when it takes it.
+func UseExec(ctx *cli.Context) {
+	execCliAndExit(ctx, useExec)
+}
+
+func useExec(ctx *cli.Context) (bool, error) {
+	if len(ctx.Args()) < 2 {
+		return true, errorInvalidArgCount(len(ctx.Args()), 2, ctx.Args())
+	}
+
+	policy, volume, err := splitVolume(ctx)
+	if err != nil {
+		return true, err
+	}
+
+	cfg, err := config.NewTopLevelConfig(ctx.GlobalString("prefix"), ctx.GlobalStringSlice("etcd"))
+	if err != nil {
+		return false, err
+	}
+
+	vc := &config.VolumeConfig{
+		PolicyName: policy,
+		VolumeName: volume,
+	}
+
+	host, err := os.Hostname()
+	if err != nil {
+		return false, err
+	}
+
+	um := &config.UseMount{
+		Volume:   vc,
+		Reason:   lock.ReasonMaintenance,
+		Hostname: host,
+	}
+
+	us := &config.UseSnapshot{
+		Volume: vc,
+		Reason: lock.ReasonMaintenance,
+	}
+
+	args := ctx.Args()[1:]
+	if args[0] == "--" {
+		if len(args) < 2 {
+			return true, fmt.Errorf("You must supply a command to run")
+		}
+		args = args[1:]
+	}
+
+	err = lock.NewDriver(cfg).ExecuteWithMultiUseLock([]config.UseLocker{um, us}, true, 0, func(ld *lock.Driver, uls []config.UseLocker) error {
```

Comment:
just wondering, should be pick the timeout from `GlobalConfig` instead of fixing it to zero?